azure/packages/test/scenario-runner/src/DocLoaderRunner.ts

```diff
@@ -231,35 +106,46 @@ export class DocLoaderRunner extends TypedEventEmitter<IRunnerEvents> implements
 		return container;
 	}
 
-	public stop(): void {}
-
-	public getStatus(): IRunnerStatus {
-		return {
-			status: this.status,
-			description: this.description(),
-			details: {},
-		};
+	protected runCore(config: IRunConfig, info: { clientIndex: number }): DocLoaderRunConfig {
+		return this.buildScenarioRunConfig(config, {
+			childId: info.clientIndex,
+			docId: this.scenarioConfig.docIds[info.clientIndex],
+			isSync: false,
+		});
 	}
 
-	private description(): string {
-		return `This stage loads a list of documents, given their IDs`;
+	protected async runSyncCore(
+		config: IRunConfig,
+		info: { clientIndex: number },
+	): Promise<IFluidContainer> {
+		return DocLoaderRunner.execRun(
+			this.buildScenarioRunConfig(config, {
+				childId: info.clientIndex,
+				docId: this.scenarioConfig.docIds[info.clientIndex],
+				isSync: true,
+			}),
+		);
 	}
 
-	private async createChild(childArgs: string[]): Promise<boolean> {
-		const envVar = { ...process.env };
-		const runnerProcess = child_process.spawn("node", childArgs, {
-			stdio: ["inherit", "inherit", "inherit", "ipc"],
-			env: envVar,
-		});
+	protected buildScenarioRunConfig(
+		runConfig: IRunConfig,
+		options: { childId: number; docId: string; isSync?: boolean },
+	): DocLoaderRunConfig {
+		const scenarioRunConfig: DocLoaderRunConfig = {
+			...runConfig,
+			childId: options.childId,
+			docId: options.docId,
+			schema: this.scenarioConfig.schema,
+			client: this.scenarioConfig.client,
+		};
+		if (!options.isSync) {
+			delete scenarioRunConfig.logger;
+			delete scenarioRunConfig.client;
```

Comment:
I get why you are doing it, just curious if there is maybe a cleaner way. I have not looked in-depth the full path, but in async scenarios we maybe need to make sure that client/logger are never passed down. This is minor anyways.